### PR TITLE
Rename package.json to clib.json

### PR DESCRIPTION
**Pull request by Aloz1**
The clib package manager has deprecated package.json in favor of clib.json (probably to reduce confusion between clib and other package managers, e.g. npm). As far as I can tell, there's no difference in file format, only in name.

### CODE REVIEW

**Comment by Aloz1** [comment]
Note, it seems that irrespective of this change, acquiring nukular via clib is currently broken.

It seems that clib now expects a tag to exist for the version specified in package.json/clib.json.

PR #287 will need to be merged to un-break clib support.

**Review by RobLoach — APPROVED** [review]
Seems appropriate.

**Comment by dumblob** [comment]
Yep, to me too. I wouldn't bump the version as it's only repository-related and not Nuklear (the library/code) related. What do you think?

E.g. if you have a counter example (because of some build tool or whatever dependency), then I'm fine merging this with the bump :wink:.

File: src/CHANGELOG [hunk 1]
```diff
@@ -8,6 +8,7 @@
 ///    - [yy]: Minor version with non-breaking API and library changes
 ///    - [zz]: Bug fix version with no direct changes to API
 ///
+/// - 2021/11/16 (4.09.1) - Renamed package.json to clib.json
```
Comment (RobLoach) [inline src/CHANGELOG:11]:
``` suggestion
```

File: clib.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "nuklear",
-  "version": "4.09.0",
+  "version": "4.09.1",
```
Comment (RobLoach) [inline clib.json:3]:
``` suggestion
  "version": "4.09.0",
```

**Review by RobLoach — CHANGES_REQUESTED** [review]
Removing the version bump in clib.json

**Comment by Aloz1** [comment]
No counter examples or objections. Have reverted version bump in 8052d27.

File: clib.json (hunk 2 above)
Comment (Aloz1) [reply clib.json]: Fixed in 8052d27

File: src/CHANGELOG (hunk 1 above)
Comment (Aloz1) [reply src/CHANGELOG]: Fixed in 8052d27

File: Readme.md [hunk 3]
```diff
@@ -185,8 +185,8 @@ Reviewing changes to `src/*` and `nuklear.h`:
   * Variable/function name casing.
   * Indentation.
   * Curly bracket (`{}`) placement.
-* Ensure that the contributor has bumped the appropriate version in
-  [package.json](https://github.com/Immediate-Mode-UI/Nuklear/blob/master/package.json)
+* Ensure that the contributer has bumped the appropriate version in
```
Comment (RobLoach) [inline Readme.md:188]:
``` suggestion
* Ensure that the contributor has bumped the appropriate version in
```

**Pull request merged**